presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -130,4 +144,29 @@ class AgendaViewModel @Inject constructor(
             }
         }
     }
+
+    private fun syncAgendaItems() {
+        agendaSyncObserver = Observer { workInfo ->
+            if(workInfo.state == WorkInfo.State.SUCCEEDED) {
+                /* Delete the ids from the deleted sync table */
+                viewModelScope.launch {
+                    agendaLocalRepository.deleteEventSyncType(SyncAgendaType.DELETE)
+                }
+            }
+        }
+
+        val syncAgendaId = viewModelScope.async {
+            syncAgendaItems.sync()
+        }
```

Comment:
I call this in `async` block as I want to wait for the result of the UUID from the worker